components/org.wso2.identity.outbound.adapter.websubhub/src/main/java/org/wso2/identity/outbound/adapter/websubhub/util/WebSubHubAdapterUtil.java

```diff
@@ -162,8 +164,15 @@ public static void makeAsyncAPICall(SecurityEventTokenPayload securityEventToken
         String jsonString;
         try {
             jsonString = mapper.writeValueAsString(securityEventTokenPayload);
-            request.setEntity(new StringEntity(jsonString));
-        } catch (IOException e) {
+            JSONParser jsonParser = new JSONParser();
+            JSONObject payloadJSON = (JSONObject) jsonParser.parse(jsonString);
+            // Encrypt the event object in the payload.
+            if (WebSubHubAdapterDataHolder.getInstance().getAdapterConfiguration().isEncryptionEnabled()) {
+                payloadJSON.put(PAYLOAD_EVENT_JSON_KEY, EventPayloadCryptographyUtils.encryptEventPayload(
+                        payloadJSON.get(PAYLOAD_EVENT_JSON_KEY).toString(), tenantDomain));
+            }
+            request.setEntity(new StringEntity(payloadJSON.toString()));
```

Comment:
if the encryption is disabled, then we don't need to use jsonParser, so better to move it inside the `if`. wdyt?
```suggestion
            // Encrypt the event object in the payload.
            if (WebSubHubAdapterDataHolder.getInstance().getAdapterConfiguration().isEncryptionEnabled()) {
                JSONParser jsonParser = new JSONParser();
                JSONObject payloadJSON = (JSONObject) jsonParser.parse(jsonString);
                payloadJSON.put(PAYLOAD_EVENT_JSON_KEY, EventPayloadCryptographyUtils.encryptEventPayload(
                        payloadJSON.get(PAYLOAD_EVENT_JSON_KEY).toString(), tenantDomain));
                jsonString = payloadJSON.toString();
            }
            request.setEntity(new StringEntity(jsonString));
```